### PR TITLE
adds Facebook and Twitter share buttons

### PR DESCRIPTION
**Pull request by skuiry**
Testing for link previews for Facebook and Twitter.

### CODE REVIEW

**Comment by malsf21** [comment]
@skuiry you should be good to merge in Getty's changes from `master` and then wrap up this PR!

**Comment by malsf21** [comment]
This is on my radar! Will review in a bit!

File: _layouts/post.html [hunk 1]
```diff
@@ -1,10 +1,22 @@
 ---
 layout: default
+image:
+  path: /img/posts/{{page.img}}"
+  height: 100
+  width: 100
 ---
 
 <article class="post-container">
+  <meta property="twitter:image" content= "/img/posts/{{page.img}}" >
```
Comment (malsf21) [inline _layouts/post.html:10]: We should prepend this url with `{{site.baseurl}}`

File: _layouts/post.html [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+  <meta property="og:image" content= "https://teachla.uclaacm.com/img/posts/default.png" >
```
Comment (malsf21) [inline _layouts/post.html:11]: Instead of `https://teachla.uclaacm.com`, let's use `{{site.url}}`

File: _layouts/post.html [hunk 3]
```diff
@@ -37,8 +49,20 @@ <h1 class="title page-title">{{page.title}}</h1>
     <div class="generated-content">
         {{content}}
     </div>
+
+<hr class="divider" />
 </article>
 
+
+<div class = "social-buttons" style="text-align: center">
```
Comment (malsf21) [inline _layouts/post.html:57]: Instead of using `style="text-align: center"`, let's bundle that in to the `.social-buttons` class.

File: _layouts/post.html [hunk 4]
```diff
@@ -37,8 +49,20 @@ <h1 class="title page-title">{{page.title}}</h1>
...
+Enjoyed the read? Share on social media! <br>
+<a class="fb-share-button"  data-layout="button" data-size="large" target="_blank" href="" class="fb-xfbml-parse-ignore">Share</a>
```
Comment (malsf21) [inline _layouts/post.html:59]: Is there supposed to be a blank `href` here?

File: _layouts/post.html [hunk 5]
```diff
@@ -37,8 +49,20 @@ <h1 class="title page-title">{{page.title}}</h1>
     <div class="generated-content">
         {{content}}
     </div>
+
+<hr class="divider" />
 </article>
 
+
+<div class = "social-buttons" style="text-align: center">
+Enjoyed the read? Share on social media! <br>
```
Comment (malsf21) [inline _layouts/post.html:58]: Let's use `<br />` (minor style thing)

File: _sass/_post.scss [hunk 6]
```diff
@@ -118,9 +118,24 @@
...
-
-
 .post-img {
     padding: 2%;
     max-width: 96%;
 }
+
+.fb-share-button {
+  text-align: center;
+  display: inline-block;
```
Comment (malsf21) [inline _sass/_post.scss:128]: This is duplicated code between `.twitter-share-button`, and maybe could be abstracted to the .social-buttons class!

File: _sass/_post.scss [hunk 7]
```diff
@@ -118,9 +118,24 @@
...
+}
+
+.twitter-share-button {
+  text-align: center;
+  display: inline-block;
+}
+
+.social-buttons{
+  margin: 0 auto;
+  width: 50%;
```
Comment (malsf21) [inline _sass/_post.scss:139]: Is this to center the contents? A flex container may be more responsive

File: _layouts/post.html [hunk 8]
```diff
@@ -64,18 +88,19 @@ <h2 class="title" id="authors">Author{% if page.authors.size > 1 %}s{% endif %}<
...
+
+        {% include blog-author-card.html name=author_name role=author_role img=author_img url=author_url %}
         {% if author.pronouns %}
             {% capture author_pronouns %}
                     {{author.pronouns}}
             {% endcapture %}
         {% else %}
             {% capture author_pronouns %}{% endcapture %}
         {% endif %}
-        {% include blog-author-card.html name=author_name role=author_role img=author_img url=author_url pronouns=author_pronouns %}
```
Comment (malsf21) [inline _layouts/post.html:74]: Similarly, I think you should undo this change - you picked the wrong side of the merge conflict here (you've disabled author pronouns now)

File: _layouts/post.html [hunk 9]
```diff
@@ -1,10 +1,22 @@
...
 
 <article class="post-container">
+  <meta property="twitter:image" content= "/img/posts/{{page.img}}" >
+  <meta property="og:image" content= "https://teachla.uclaacm.com/img/posts/default.png" >
     <h1 class="title page-title">{{page.title}}</h1>
+    <!-- required for Facebook "Share" button -->
+    <div id="fb-root"></div>
+    <script async defer crossorigin="anonymous" src="https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v9.0" nonce="VotTmBal"></script>
+
+
```
Comment (malsf21) [inline _layouts/post.html:17]: From the FB documentation:
>  Include the JavaScript SDK on your page once, ideally right after the opening body tag.

We want the script tag to load as quickly as possible, since we're making the request in the background as the rest of the document/tags are parsed. 
* at the very least, we should move it to the top of the layout
* something you can consider: maybe modifying `_layout/default.html` to conditionally load this, but that's overkill for now probably

File: _layouts/post.html [hunk 10]
```diff
@@ -37,8 +49,20 @@ <h1 class="title page-title">{{page.title}}</h1>
...
     </div>
+
+<hr class="divider" />
 </article>
 
+
+<div class = "social-buttons" style="text-align: center">
+Enjoyed the read? Share on social media! <br>
+<a class="fb-share-button"  data-layout="button" data-size="large" target="_blank" href="" class="fb-xfbml-parse-ignore">Share</a>
+<a class="twitter-share-button" href="https://twitter.com/share?ref_src=twsrc%5Etfw" data-size="large" data-show-count="false">Tweet</a><script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```
Comment (malsf21) [inline _layouts/post.html:60]: Siimlar comment on the script tag as above; also can we `defer` it? Not sure if that causes problems!

**Review by malsf21 — CHANGES_REQUESTED** [review]
Hey Shounak, 

Thanks for submitting the PR! Excited to see things come together, though I have a few comments for you - the general gist:

* I think you made some mistakes resolving a merge conflict, and you've introduced some regressions from previous features - make sure to revert those!
* discussed a few ways to clean up the HTML & CSS
* there are some situations where we should use Jekyll variables
* we may want to preload the script tags earlier - if they're `async defer`, they aren't render blocking, so the earlier we load them the faster they resolve!

Functionality-wise, it looks like both share options (including the image previews) work as intended - great work there 😄

File: _layouts/post.html (hunk 4 above)
Comment (skuiry) [reply _layouts/post.html:59]: Yep! The href is appended to the site's url which is odd but that's how the script functions. When I put anything here, it would append it to the post's url which led to a 404 error. By keeping it empty, the link redirects to the blog post.

File: _layouts/team-page.html [hunk 11]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (malsf21) [inline _layouts/team-page.html:1]: Should this file exist? I think you may have picked the wrong outcome of the merge conflict resolver.

**Review by malsf21 — APPROVED** [review]
Great work @skuiry, thanks for all the investigation and back-and-forth you've done with this PR! Really excited to see it merge 😊
